templates/lib/yarp/serialize.rb.erb

```diff
@@ -157,11 +157,16 @@ module YARP
 
         unless constant
           offset = constant_pool_offset + index * 8
-
           start = serialized.unpack1("L", offset: offset)
           length = serialized.unpack1("L", offset: offset + 4)
 
-          constant = input.byteslice(start, length).to_sym
+          constant =
+            if (start & (1 << 31)) == 0
```

Comment:
Maybe a good use-case for Integer#anybits?/nobits?

Reply:
Updated